vocabulary-draft/source-code-version-control.md

```diff
@@ -0,0 +1,27 @@
+# Source Code Version Control Events
+
+This phase includes events related to changes in Source Code repositories that are relevant from a Continuous Delivery perspective.
+
+
+## Repository Events
+
+These events are related to Source Code repositories
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+- **Change Committed**: a change is commited to a branch
```

Comment:
When using Gerrit we could have the following scenario:

- Write your code, commit it and push your changes to the refs/for/<branch name> (New change-id)
- Review of the code
- Update to fix the comments (new commit)
- Approve the change and merge it

When would we send the **Change Committed** event? At each push or one event for the whole change?

Reply:
@m-linner-ericsson this is up for discussion, but the original intent that I was trying to capture is when you push a commit (a batch of changes) to a branch.

Reply:
@salaboy , I guess you mean the action that in GitHub and Gitlab is called _merge_ and in Gerrit is called _submit_ (as stated here: https://github.com/cdfoundation/sig-interoperability/blob/master/docs/vocabulary.md#scm-tools-and-technologies) or? This would be the natural event to kick off a CI pipeline on.
@m-linner-ericsson , I think you're requesting an event stating that a new commit has been added for review (called patch-set in Gerrit and PR/MR commit in GitHub/Gitlab), right? In Ericsson we often kick off _pre-merge_ pipelines based on such events, to automatically run any tests on the commit prior to merging/submitting it.

Reply:
My comment from the old PR:
> For me, the "Change being proposed as in a Pull Request" would correspond to Change Created (and Change Updated if new commits are added to the PR), and "Change being merged to a branch" would correspond to Change Merged.

I should clarify for Change Merged that I mean merged to the target branch for the review, not to the "review branch"

Reply:
This is super useful feedback, I think that now we are close to defining the terms that we need, I am not strongly attached to any of those terms, I think that choosing one that is widely recognized in the industry will make it easier to adopt that brand new terms defined by us, so I will tend to go for GitHub/Gitlab terms. Do you have any preferences?